ethereum/chain.py

```diff
@@ -263,21 +264,21 @@ def add_block(self, block):
                         break
                     b = self.get_parent(b)
                 replace_from = b.header.number
-                for i in xrange(replace_from, 2**63 - 1):
+                for i in islice(count(), replace_from, 2**63 - 1):
```

Comment:
Oh, I think you're using `islice` for the upper bound part of this.  

Reading this code, the `2**63-1` appears to just be a true upper bound for block height that we never actually expect to reach.  I think it's safe to instead just do `count(replace_from)` and count upwards indefinitely since counting upwards to `2**63-1` is effectively going to be a loop that will never complete.